app/src/org/commcare/gis/MapboxLocationPickerActivity.kt

```diff
@@ -0,0 +1,206 @@
+package org.commcare.gis
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.ViewModelProvider
+import com.mapbox.android.core.location.LocationEngineCallback
+import com.mapbox.android.core.location.LocationEngineResult
+import com.mapbox.mapboxsdk.camera.CameraPosition
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory
+import com.mapbox.mapboxsdk.geometry.LatLng
+import com.mapbox.mapboxsdk.location.LocationComponentActivationOptions
+import com.mapbox.mapboxsdk.location.modes.CameraMode
+import com.mapbox.mapboxsdk.location.modes.RenderMode
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.mapboxsdk.maps.Style
+import com.mapbox.mapboxsdk.plugins.annotation.Symbol
+import com.mapbox.mapboxsdk.plugins.annotation.SymbolManager
+import com.mapbox.mapboxsdk.plugins.annotation.SymbolOptions
+import kotlinx.android.synthetic.main.activity_mapbox_location_picker.*
+import org.commcare.activities.components.FormEntryConstants
+import org.commcare.dalvik.R
+import org.commcare.utils.GeoUtils
+import org.commcare.views.widgets.GeoPointWidget
+
+class MapboxLocationPickerActivity : BaseMapboxActivity() {
+
+    companion object {
+        const val MARKER_ICON_IMAGE_ID = "marker_icon_image"
+    }
+
+    private lateinit var viewModel: MapboxLocationPickerViewModel
+    private var loadedStyle: Style? = null
+    private val mapStyles = arrayOf(
```

Comment:
we support 4 styles for the gmaps based map. Should we add the other two here as well ?